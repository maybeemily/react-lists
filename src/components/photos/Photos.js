import React from 'react';
import PropTypes from 'prop-types';
import Photo from './Photo';

function Photos({ photoArray }) {
  const photoList = photoArray.map(photo => {
    return (
      <li key={photo}>
        <Photo photo={photo}/>
      </li>
    );
  });

  return (
    <ul>
      {photoList}
    </ul>
  );
}

Photos.propTypes = {
  photoArray: PropTypes.array.isRequired
};

export default Photos;
