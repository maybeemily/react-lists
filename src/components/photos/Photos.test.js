import React from 'react';
import { shallow } from 'enzyme';
import Photos from './Photos';

describe('Photos component', () => {
  it('renders the photo component', () => {
    const photoArray = [
      '/img.jpg',
      './img2.jpg',
      './img3.jpg'
    ];
    const wrapper = shallow(<Photos photoArray={photoArray} />);
    expect(wrapper).toMatchSnapshot();
  });
});
