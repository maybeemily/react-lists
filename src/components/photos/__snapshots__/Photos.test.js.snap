// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos component renders the photo component 1`] = `
<ul>
  <li
    key="/img.jpg"
  >
    <Photo
      photo="/img.jpg"
    />
  </li>
  <li
    key="./img2.jpg"
  >
    <Photo
      photo="./img2.jpg"
    />
  </li>
  <li
    key="./img3.jpg"
  >
    <Photo
      photo="./img3.jpg"
    />
  </li>
</ul>
`;
